packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -247,7 +247,7 @@ class AttachCommand extends FlutterCommand {
             portForwarder: device.portForwarder,
           );
           printStatus('Waiting for a connection from Flutter on ${device.name}...');
-          observatoryUri = observatoryDiscovery.uris;
+          observatoryUri = Stream<Uri>.fromFuture(observatoryDiscovery.uris.first).asBroadcastStream();
```

Comment:
Won't a broadcast stream have timing issues? Events aren't buffered, if the subscriber is added after discovery it would be missed